src/Arc4u.Standard.OAuth2.AspNetCore.Authentication/Extensions/AuthenticationExtensions.cs

```diff
@@ -185,63 +186,72 @@ public static AuthenticationBuilder AddOidcAuthentication(this IServiceCollectio
         }
+
         if (string.IsNullOrWhiteSpace(settings.OAuth2SettingsSectionPath))
         {
-            configErrors += "We need a setting section to configure OAuth2." + System.Environment.NewLine;
+            configErrorsStringBuilder.AppendLine("We need a setting section to configure OAuth2.");
         }
+
         if (string.IsNullOrWhiteSpace(settings.CertificateSectionPath))
         {
-            configErrors += "We need a setting section to specify the certificate to protect your sensitive information." + System.Environment.NewLine;
+            configErrorsStringBuilder.AppendLine("We need a setting section to specify the certificate to protect your sensitive information.");
         }
+
         if (string.IsNullOrWhiteSpace(settings.DataProtectionSectionPath))
         {
-            configErrors += "We need a setting section to configure the DataProtection cache store." + System.Environment.NewLine;
+            configErrorsStringBuilder.AppendLine("We need a setting section to configure the DataProtection cache store.");
         }
+
         if (string.IsNullOrWhiteSpace(settings.JwtBearerEventsType))
         {
-            configErrors += "The JwtBearerEventsType must be defined." + System.Environment.NewLine;
+            configErrorsStringBuilder.AppendLine("The JwtBearerEventsType must be defined.");
         }
+
         if (string.IsNullOrWhiteSpace(settings.ClaimsIdentifierSectionPath))
         {
-            configErrors += "We need a setting section to specify the claims used to identify a user." + System.Environment.NewLine;
+            configErrorsStringBuilder.AppendLine("We need a setting section to specify the claims used to identify a user.");
         }
 
-        if (configErrors is not null)
+        if (string.IsNullOrWhiteSpace(settings.CookieAuthenticationEventsType))
         {
-            throw new ConfigurationException(configErrors);
+            configErrorsStringBuilder.AppendLine("The CookieAuthenticationEventsType must be defined.");
         }
 
-        var jwtBearerEventsType = Type.GetType(settings.JwtBearerEventsType, false);
+        if (string.IsNullOrWhiteSpace(settings.OpenIdConnectEventsType))
+        {
+            configErrorsStringBuilder.AppendLine("The OpenIdConnectEventsType must be defined.");
+        }
 
-        if (string.IsNullOrWhiteSpace(settings.CookieAuthenticationEventsType))
+        if (string.IsNullOrWhiteSpace(settings.ResponseType))
         {
-            throw new MissingFieldException("The CookieAuthenticationEventsType must be defined.");
+            configErrorsStringBuilder.AppendLine("A ResponseType is mandatory to define the OpenId Connect protocol.");
         }
-        var cookieAuthenticationEventsType = Type.GetType(settings.CookieAuthenticationEventsType, true);
 
-        if (string.IsNullOrWhiteSpace(settings.OpenIdConnectEventsType))
```

Comment:
Instead of throwining the exception at the end if the setting is missing
I have moved up the check so the error is added to the error list.

Change is that MissingFieldException would not e thrown anymore.